src/runtime/math.py

```diff
@@ -687,6 +688,7 @@ def log1p(x: f64) -> f64:
     return log(1.0 + x)
 
 
+@overload
```

Comment:
This is kind of hacky and shouldn't we use some name tweaks at the ASR level?

Reply:
I guess overload just does that. We can do it by hand but that will be needed only for `fmod` and `log` and nothing else. So ASR name tweaks will be hackish as well. The problem is the `log` which is called in `lfortran_intrinsics.h` is defined in our generated C code as well and hence gets linked to our definition instead of the one in `cmath`. That’s what causes infinite call loop leading to segmentation fault. Now `overload` automatically does name mangling and hence linker correctly links because now `log` is not present instead its as present as a mangled name. Doing the same mangling by hand in ASR is more hackish. Also overloaded  `log`/`fmod` are harmless as far as I can see.

Reply:
The other way is to do it for every imported symbol. For example, `from math import log, fabs` should be stored as `math_log__external_`, `math_fabs__external_`. Then it will be uniform for every imported symbol, so no hacks then. However, I am not sure that's worth the effort. @certik Do you have any other approach in mind?

Reply:
> The other way is to do it for every imported symbol. For example, `from math import log, fabs` should be stored as `math_log__external_`, `math_fabs__external_`. Then it will be uniform for every imported symbol, so no hacks then

This sounds better to me. The reason is that the users might not be aware of this and may end up writing a function without an `overload` decorator (and it's quite intuitive not to use `overload` for a single function definition.) We need to figure out a generalized approach to handle this and the above might be the one.

Reply:
I did a more sensible change (by implementing log for f32, fmod for (f32, f32) and log1p for f32). So the overloads make sense now. 



> The other way is to do it for every imported symbol. For example, `from math import log, fabs` should be stored as `math_log__external_`, `math_fabs__external_`.

This is not possible actually because call names won't resolve correctly. This is a problem on our compiler's end and I fixed it by implementing, log and fmod correctly.